modulemd/modulemd-eol.c

```diff
@@ -0,0 +1,1196 @@
+
+  g_clear_pointer (&self->module_name, g_free);
+  self->module_name = g_strdup (module_name);
+
+  g_object_notify_by_pspec (G_OBJECT (self), properties[PROP_MODULE_NAME]);
+}
+
+
+static void
+modulemd_eol_set_module_stream (ModulemdEol *self, const gchar *module_stream)
+{
+  g_return_if_fail (MODULEMD_IS_EOL (self));
+  g_return_if_fail (module_stream);
+  g_return_if_fail (g_strcmp0 (module_stream, O_DEFAULT_STRING));
+
+  g_clear_pointer (&self->module_stream, g_free);
+  self->module_stream = g_strdup (module_stream);
+
+  g_object_notify_by_pspec (G_OBJECT (self), properties[PROP_MODULE_STREAM]);
+}
+
+
+void
+modulemd_eol_set_module_context (ModulemdEol *self,
+                                 const gchar *module_context)
+{
+  g_return_if_fail (MODULEMD_IS_EOL (self));
+
+  g_clear_pointer (&self->module_context, g_free);
+  self->module_context = g_strdup (module_context);
+
+  g_object_notify_by_pspec (G_OBJECT (self), properties[PROP_MODULE_CONTEXT]);
+}
+
+void
+modulemd_eol_set_eol_date (ModulemdEol *self, guint64 eol_date)
+{
+  g_return_if_fail (MODULEMD_IS_EOL (self));
+  g_return_if_fail (!(eol_date && self->reset));
+
+  self->eol_date = eol_date;
+
+  g_object_notify_by_pspec (G_OBJECT (self), properties[PROP_EOL_DATE]);
+}
+
+
+void
+modulemd_eol_set_message (ModulemdEol *self, const gchar *message)
+{
+  g_return_if_fail (MODULEMD_IS_EOL (self));
+  g_return_if_fail (message);
+  g_return_if_fail (g_strcmp0 (message, O_DEFAULT_STRING));
+
+  g_clear_pointer (&self->message, g_free);
+  self->message = g_strdup (message);
+
+  g_object_notify_by_pspec (G_OBJECT (self), properties[PROP_MESSAGE]);
+}
+
+void
+modulemd_eol_set_obsoleted_by_module_name (
+  ModulemdEol *self, const gchar *obsoleted_by_module_name)
+{
+  g_return_if_fail (MODULEMD_IS_EOL (self));
+  g_return_if_fail (!(obsoleted_by_module_name && self->reset));
+
+  g_clear_pointer (&self->obsoleted_by_module_name, g_free);
+  self->obsoleted_by_module_name = g_strdup (obsoleted_by_module_name);
+
+  g_object_notify_by_pspec (G_OBJECT (self),
+                            properties[PROP_OBSOLETED_BY_MODULE_NAME]);
+}
+
+void
+modulemd_eol_set_obsoleted_by_module_stream (
+  ModulemdEol *self, const gchar *obsoleted_by_module_stream)
+{
+  g_return_if_fail (MODULEMD_IS_EOL (self));
+  g_return_if_fail (!(obsoleted_by_module_stream && self->reset));
+
+  g_clear_pointer (&self->obsoleted_by_module_stream, g_free);
+  self->obsoleted_by_module_stream = g_strdup (obsoleted_by_module_stream);
+
+  g_object_notify_by_pspec (G_OBJECT (self),
+                            properties[PROP_OBSOLETED_BY_MODULE_STREAM]);
+}
+
+void
+modulemd_eol_set_obsoleted_by (ModulemdEol *self,
+                               const gchar *obsoleted_by_module_name,
+                               const gchar *obsoleted_by_module_stream)
+{
+  g_return_if_fail (MODULEMD_IS_EOL (self));
+  g_return_if_fail (obsoleted_by_module_name);
+  g_return_if_fail (obsoleted_by_module_stream);
```

Comment:
These are redundant and also should probably not `abort()` if they fail. Just check that they are valid in `modulemd_eol_validate()`.

Reply:
I have removed the checks, but since these values are optional they should not be checked in `modulemd_eol_validate()` right? They are always valid.

Reply:
In `modulemd_eol_validate()` you should at least check that they are either both present or both absent. Having just one or the other doesn't make sense (and that should be mentioned in the spec too).

Reply:
Added